pkg/types/time.go

```diff
@@ -1524,29 +1524,28 @@ func (d Duration) ConvertToYearFromNow(ctx Context, now gotime.Time) (int64, err
 	return AdjustYear(int64(datePart.Year()), false)
 }
 
+// TruncateFrac truncates the fractional second precision
+// and returns a new Duration.
+// so 10:10:10.999999 truncate 0 -> 10:10:10
+// and 10:10:10.66666 truncate 3 -> 10:10:10.666
+func (d Duration) TruncateFrac(fsp int) (Duration, error) {
+	fsp, err := CheckFsp(fsp)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	return Duration{Duration: d.Truncate(gotime.Duration(math.Pow10(9-fsp)) * gotime.Nanosecond), Fsp: fsp}, nil
+}
+
 // RoundFrac rounds fractional seconds precision with new fsp and returns a new one.
 // We will use the “round half up” rule, e.g, >= 0.5 -> 1, < 0.5 -> 0,
 // so 10:10:10.999999 round 0 -> 10:10:11
 // and 10:10:10.000000 round 0 -> 10:10:10
-func (d Duration) RoundFrac(fsp int, loc *gotime.Location) (Duration, error) {
-	tz := loc
-	if tz == nil {
-		logutil.BgLogger().Warn("use gotime.local because sc.timezone is nil")
-		tz = gotime.Local
-	}
-
+func (d Duration) RoundFrac(fsp int) (Duration, error) {
```

Comment:
Duration should not use gotime.Time for rounding or truncating. There is simply no need or benefit to convert back and forth.